src/sdk/namespace_fs.js

```diff
@@ -293,11 +296,21 @@ const versions_dir_cache = new LRUCache({
                 usage += ent.name.length + 4;
             }
         }
-        return { time, stat, sorted_entries, usage };
+        return { time, stat, ver_dir_stat, sorted_entries, usage };
     },
-    validate: async ({ stat }, { dir_path, fs_context }) => {
+    validate: async ({ stat, ver_dir_stat }, { dir_path, fs_context }) => {
         const new_stat = await nb_native().fs.stat(fs_context, dir_path);
-        return (new_stat.ino === stat.ino && new_stat.mtimeNsBigint === stat.mtimeNsBigint);
+        if (ver_dir_stat) {
```

Comment:
Let me present my thoughts here.

Base of the thought: In any case, when we update the object, will will end up calling `load()` which will eventually load objects and its versions.

case 1: No versions exist, only latest object CREATED: Else part will be executed (because `ver_dir_stat` can't be populated from the cache) and it will be false, so load() will be called 

case 2: version of the object CREATED, So object created in case 1 will be moved to `.versions`: Else part will be executed (because `ver_dir_stat` can't be populated from the cache) and it will return false since the objects have been modified under main dir and returns false and lead to call to `load()`

case 3: 2nd version of the object CREATED: In this `mtime` is changed on both main dir and `.versions` dir, we will end up calling if() in this case and returns false since objects have been modified and lead to call to `load()`

case 4: If we delete any version of the object under `.versions`,  then we call list_versions(), then if part will be executed because `ver_dir_stat` will be already there in the cache and returns false since the dir has been modified

`if(ver_dir_stat)` condition addresses only those cases (like case 4) in which `mtime` modification only happens to `.versions` and not  to main directory. Whenever we PUT the object both main dir and `.versions` dir's `mtime` will be modified because we make the movement of the objects between main dir and `.versions` dir

Please correct me if I am missing anything

Reply:
hmm got you ok